src/main/java/org/aksw/simba/lemming/ColouredGraphDecorator.java

```diff
@@ -0,0 +1,300 @@
+/**
+ * 
+ */
+package org.aksw.simba.lemming;
+
+import org.aksw.simba.lemming.mimicgraph.constraints.TripleBaseSingleID;
+import org.aksw.simba.lemming.util.IntSetUtil;
+
+import com.carrotsearch.hppc.BitSet;
+
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntSet;
+
+/**
+ * Base Decorator for Coloured Graph Class
+ * 
+ * @author Pranav
+ */
+public class ColouredGraphDecorator implements IColouredGraph {
+
+    protected IColouredGraph graph;
+    protected boolean isAddingEdge;
+    protected TripleBaseSingleID triple;
+
+    public ColouredGraphDecorator() {
+        this.graph = null;
+        this.isAddingEdge = false;
+        this.triple = null;
+    }
```

Comment:
Does a decorator that does not have anything to decorate really make sense?

Reply:
Removed.